Support New MDAnalysis Timestep Namespace

### Description

The namespace of the `MDAnalysis` object that we accept as system-like has changed with their 2.3.0 release. Previously it was `MDAnalysis.coordinates.base.Timestep`, it is now `MDAnalysis.coordinates.timestep.Timestep`

### Steps to Reproduce

```python
Just run the tests with mdanalysis==2.3.0

$ pytest freud/tests/integration/test_reader_integrations.py -k test_mdanalysis_gsd
```


### Error Output

```shell
=========================================================================================================================================================================== test session starts ============================================================================================================================================================================
platform darwin -- Python 3.9.13, pytest-7.1.3, pluggy-1.0.0
rootdir: /Users/tomwalt/code/freud, configfile: setup.cfg
collected 6 items / 5 deselected / 1 selected

freud/tests/integration/test_reader_integrations.py F                                                                                                                                                                                                                                                                                                                [100%]

================================================================================================================================================================================= FAILURES =================================================================================================================================================================================
________________________________________________________________________________________________________________________________________________________________ TestReaderIntegrations.test_mdanalysis_gsd ________________________________________________________________________________________________________________________________________________________________

self = <test_reader_integrations.TestReaderIntegrations object at 0x11651ea90>

    @pytest.mark.skipif(
        sys.platform.startswith("win"), reason="Not supported on Windows."
    )
    def test_mdanalysis_gsd(self):
        MDAnalysis = pytest.importorskip("MDAnalysis")
        reader = MDAnalysis.coordinates.GSD.GSDReader(LJ_GSD)
>       self.run_analyses(reader)

freud/tests/integration/test_reader_integrations.py:36:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
freud/tests/integration/test_reader_integrations.py:26: in run_analyses
    rdf.compute(system, reset=False)
util.pyx:165: in freud.util._Compute.__getattribute__.compute_wrapper
    ???
density.pyx:676: in freud.density.RDF.compute
    ???
locality.pyx:925: in freud.locality._PairCompute._preprocess_arguments
    ???
locality.pyx:371: in freud.locality.NeighborQuery.from_system
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: __cinit__() takes exactly 2 positional arguments (1000 given)

locality.pyx:804: TypeError
========================================================================================================================================================================= short test summary info ==========================================================================================================================================================================
FAILED freud/tests/integration/test_reader_integrations.py::TestReaderIntegrations::test_mdanalysis_gsd - TypeError: __cinit__() takes exactly 2 positional arguments (1000 given)
```


### freud Version

v2.11.0

### Python Version

v3.9.13

### System Platform

MacOS

### Installation method

built from source

### Developer

I will submit a pull request that fixes this problem.

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -9,6 +9,9 @@ and this project adheres to
 ### Added
 * Mass dependence in `freud.cluster.ClusterProperties` and calculation of inertia tensor and center of mass.
 
+### Fixed
+* Compatibility with new namespace for `MDAnalysis.coordinates.timestep.Timestep`.
+
 ## v2.11.0 -- 2022-08-09
 
 ### Added

--- a/freud/locality.pyx
+++ b/freud/locality.pyx
@@ -286,7 +286,7 @@ cdef class NeighborQuery:
         * A sequence of :code:`(box, points)` where :code:`box` is a
           :class:`~.box.Box` and :code:`points` is a :class:`numpy.ndarray`.
         * Objects with attributes :code:`box` and :code:`points`.
-        * :class:`MDAnalysis.coordinates.base.Timestep`
+        * :class:`MDAnalysis.coordinates.timestep.Timestep`
         * :class:`gsd.hoomd.Snapshot`
         * :class:`garnett.trajectory.Frame`
         * :class:`ovito.data.DataCollection`
@@ -315,7 +315,7 @@ cdef class NeighborQuery:
             return system
 
         # MDAnalysis compatibility
-        elif _match_class_path(system, 'MDAnalysis.coordinates.base.Timestep'):
+        elif _match_class_path(system, 'MDAnalysis.coordinates.timestep.Timestep'):
             system = (system.triclinic_dimensions, system.positions)
 
         # GSD and HOOMD-blue 3 snapshot compatibility

--- a/freud/locality.pyx
+++ b/freud/locality.pyx
@@ -315,7 +315,11 @@ cdef class NeighborQuery:
             return system
 
         # MDAnalysis compatibility
-        elif _match_class_path(system, 'MDAnalysis.coordinates.timestep.Timestep'):
+        # base namespace for mdanalysis<2.3.0
+        # timestep namespace for mdanalysis>=2.3.0
+        elif _match_class_path(system,
+                               'MDAnalysis.coordinates.base.Timestep',
+                               'MDAnalysis.coordinates.timestep.Timestep'):
             system = (system.triclinic_dimensions, system.positions)
 
         # GSD and HOOMD-blue 3 snapshot compatibility

